docs/proposal/20210823-gpu-support.md

```diff
@@ -0,0 +1,200 @@
+      * [Benefits](#benefits)
+      * [Mitigations for current proposal](#mitigations-for-current-proposal)
+  * [Upgrade Strategy](#upgrade-strategy)
+  * [Additional Details](#additional-details)
+    * [Test Plan](#test-plan)
+    * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+
+## Glossary
+
+* CAPV - An abbreviation of Cluster API Provider vSphere
+
+## Summary
+
+CAPV currently doesnot support GPU based infrastructure. 
+
+This proposal outlines ways to provision GPU support in CAPV. The proposed changes will maintain backwards compatibility and maintain the existing behaviour without any extra user configuration.  
+
+## Motivation
+
+CAPV does not currently support GPU-accelerated workloads which has limited total addressable market.Competitively, all of the hyperscale cloud providers (AWS, Azure, & GCP) offer GPU-accelerated virtualization and Kubernetes platforms. Many of them additionally offer robust vertically integrated AI/ML solutions building on open source technology. GPU support is a long-requested feature. With this capability, user can leverage GPU support.
+
+### Goals
+
+1. To enable GPU Support on Vsphere Kubernetes Cluster.
+2. To allow vGPU support for multi node clusters
+3. To allow GPU Direct support for multi node clusters
+4. To allow PCI passthrough support for single node clusters.
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - using vGPU
+
+Stacy works at an IT department of a organization where AI/ML models are run every hour and the entire staff wants to leverage GPU supported nodes. VGPU's are able to provide close to bare metal performance. Stacy needs a lot of sharing GPU’s. 
+vGPU provides a full fleshed cluster from which a virtual GPU can be requested.
+
+#### Story 2 - GPU Direct implementation
+
+Tony is an Engineer at a big organization with multiple centers across the Globe. Tony needs a big GPU cluster to address the needs of his 
+organization. So they intend to use GPU Direc. GPU direct is  technically a pool of GPU’s connected by a network card in a kind of  peer to peer network. This is the latest technology and gives advantage of shared GPU’s from a pool. GPU  direct is the technology which helps in linking multiple GPU’s. Tony wants to leverage this technology so that it is easier for him to manage resource allocation.
+
+
+#### Story 3 - PCI passthrough for single node Customer
+
+Alex is an Engineer at retail organization that requires single GPU node. They use one node with GPU attached and want to keep things simple. Alex can simply add this GPU connected machine to the cluster and that should do the job. While selecting nodes, Alex can use appropriate labels to run his AI/ML workload on this particular node. PCI passthrough will provide direct GPU support. Challenges that Alex can face is that using passthrough Alex wouln't be able to migrate nodes.
```

Comment:
@MaxRink I don't get your question here. Non GPU will be counted among regular nodes in my opinion.

Reply:
I mean passthrough for PCI-E devices that are not GPUs, as PCI-E passthrough shouldnt really care if its a GPU, a NIC or even an HBA

Reply:
Yeah, If we put it that way, yes. It also applies for non GPU devices.

Reply:
@yastij Does it make sense to separate out the PCI Passthrough support to a separate proposal? This could also enable the workflow of provisioning VMs physical NICs attached to the VMs created by CAPV.

And if it is too much work to separate the proposal out, I +1 Naadir's comments on retitling the proposal since PCI passthrough support is not strictly GPU related.